### PR TITLE
Only support Linux

### PR DESCRIPTION
**Pull request by calccrypto**

### CODE REVIEW

**Comment by bwatkinson** [comment]
Let's hold off on merging this one. There are a lot of places all over the code were we still need to remove `#if` for things like `WIN32, DARWIN, AIX, FREEBSD`. Also, I have no idea what `IRIX` is, but we would look at that too. We should clean out all non-Linux related code in one commit.

File: src/base/io_buffers.c [hunk 1]
```diff
@@ -124,61 +111,31 @@ xdd_init_io_buffers(worker_data_t *wdp) {
...
-#if (IRIX || SOLARIS || LINUX || AIX || DARWIN || FREEBSD)
 		bufp = valloc(buffer_size);
-#else
-		bufp = malloc(buffer_size);
-#endif
-#endif 
 	} else { /* Allocate memory the normal way */
-#if (AIX || LINUX)
 		posix_memalign((void **)&bufp, sysconf(_SC_PAGESIZE), buffer_size);
-#elif (IRIX || SOLARIS || LINUX || DARWIN || FREEBSD)
```
Comment (calccrypto) [inline src/base/io_buffers.c:140]: Is this the right choice?

Comment (bwatkinson) [reply src/base/io_buffers.c:140]: It appears so. Interesting that it is always allocated in page sized/aligned chunks. My guess is is this due to this code also handing E2E buffers. See the comment above this function.

**Comment by bwatkinson** [comment]
I will try and get time this week to help out with this. We really need to get this merged ASAP. The more code cleanup the better at this point.

**Review by bwatkinson — CHANGES_REQUESTED** [review]
Everything looked good to me and it builds so that is good lol. However, can you go into `src/common/memory.c` and remove the whole `#ifdef notdef` block in `xdd_unlock_memory()`? It seems to allude to being some old code for AIX. However, I honestly don't know what the purpose of it is in general as I don't see where `notdef` would be even set.

**Comment by calccrypto** [comment]
Removed

**Review by bwatkinson — APPROVED** [review]

**Review by MigeljanImeri — APPROVED** [review]

**Pull request merged**